src/NuGet.Clients/NuGet.Console/WpfConsole/WpfConsole.cs

```diff
@@ -850,8 +871,7 @@ public object Content
 
             public async Task WriteProgressAsync(string operation, int percentComplete)
             {
-                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                _impl.WriteProgress(operation, percentComplete);
+                await _impl.WriteProgressAsync(operation, percentComplete);
```

Comment:
Ok. Addressed. But I recently read something interesting about having async is useful for in case of Aggregate error is thrown. Without async in signature not all task exceptions are go into aggregate error only exception from this one without async bubble up. 
I'll double check and if it's true then I'll present during `KE` day.